x-pack/plugins/ml/public/file_datavisualizer/components/fields_stats/fields_stats.js

```diff
@@ -46,38 +46,53 @@ function createFields(results) {
   const {
     mappings,
     field_stats: fieldStats,
+    column_names: columnNames,
     num_messages_analyzed: numMessagesAnalyzed,
     timestamp_field: timestampField,
   } = results;
 
   let fields = [];
 
   if (mappings && fieldStats) {
-    fields = Object.keys(fieldStats).map((fName) => {
-      const field = { name: fName };
-      const f  = fieldStats[fName];
-      const m  = mappings[fName];
-
-      // sometimes the timestamp field is not in the mappings, and so our
-      // collection of fields will be missing a time field with a type of date
-      if (fName === timestampField && field.type === undefined) {
-        field.type = 'date';
-      }
+    // if columnNames exists (i.e delimited) use it for the field list
+    // so we get the same order
+    const tempFields = (columnNames !== undefined) ? columnNames : Object.keys(fieldStats);
 
-      if (f !== undefined) {
-        Object.assign(field, f);
-      }
+    fields = tempFields.map((fName) => {
+      if (fieldStats[fName] !== undefined) {
+        const field = { name: fName };
+        const f  = fieldStats[fName];
+        const m  = mappings[fName];
```

Comment:
Just a nitpick - I think in general we probably want to have explicit variable names for clarity and readability.